recipes/dgl/conda-build.patch

```diff
@@ -0,0 +1,291 @@
++else(EXTERNAL_NANOFLANN_PATH)
++  target_include_directories(dgl PRIVATE "third_party/nanoflann/include")
++endif(EXTERNAL_NANOFLANN_PATH)
+ 
+-# Compile METIS
+-if(NOT MSVC)
+-  set(GKLIB_PATH "${CMAKE_CURRENT_SOURCE_DIR}/third_party/METIS/GKlib")
+-  include(${GKLIB_PATH}/GKlibSystem.cmake)
+-  include_directories(${GKLIB_PATH})
+-  include_directories("third_party/METIS/include/")
+-  add_subdirectory("third_party/METIS/libmetis/")
+-  list(APPEND DGL_LINKER_LIBS metis)
+-endif(NOT MSVC)
++if (USE_LIBXSMM)
++  target_include_directories(dgl PRIVATE "third_party/libxsmm/include")
++endif()
+ 
+-# Compile LIBXSMM
+-if((NOT MSVC) AND USE_LIBXSMM)
+-  set(LIBXSMM_COMPILER "${CMAKE_C_COMPILER}")
+-  if (APPLE)
+-    set(LIBXSMM_COMPILER "${CMAKE_C_COMPILER} -isysroot ${CMAKE_OSX_SYSROOT}")
++if (EXTERNAL_METIS_PATH)
++  find_package(METIS REQUIRED)
++  if (NOT METIS_FOUND)
++    message(FATAL_ERROR "Failed to find METIS library")
++  else()
++    message(STATUS "Found METIS library")
++    target_include_directories(dgl SYSTEM PUBLIC ${METIS_INCLUDE_DIR})
++    list(APPEND DGL_LINKER_LIBS ${METIS_LIBRARIES})
+   endif()
+-  if(REBUILD_LIBXSMM)
+-    add_custom_target(libxsmm COMMAND make realclean COMMAND make -j ECFLAGS="-Wno-error=deprecated-declarations" BLAS=0 CC=${LIBXSMM_COMPILER}
+-                      WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/third_party/libxsmm
+-                      )
+-  else(REBUILD_LIBXSMM)
+-    add_custom_target(libxsmm COMMAND make -j ECFLAGS="-Wno-error=deprecated-declarations" BLAS=0 CC=${LIBXSMM_COMPILER}
+-                      WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/third_party/libxsmm
+-                      )
+-  endif(REBUILD_LIBXSMM)
+-  add_dependencies(dgl libxsmm)
+-  list(APPEND DGL_LINKER_LIBS -L${CMAKE_SOURCE_DIR}/third_party/libxsmm/lib/ xsmm)
+-endif((NOT MSVC) AND USE_LIBXSMM)
++else(EXTERNAL_METIS_PATH)
++  target_include_directories(dgl PRIVATE "third_party/METIS/include")
++  # Compile METIS
++  if(NOT MSVC)
++    set(GKLIB_PATH "${CMAKE_CURRENT_SOURCE_DIR}/third_party/METIS/GKlib")
++    include(${GKLIB_PATH}/GKlibSystem.cmake)
++    include_directories(${GKLIB_PATH})
++    include_directories("third_party/METIS/include/")
++    add_subdirectory("third_party/METIS/libmetis/")
++    list(APPEND DGL_LINKER_LIBS metis)
++    endif(NOT MSVC)
++endif(EXTERNAL_METIS_PATH)
+ 
+ if(NOT MSVC)
+   # Only build tensorpipe on linux
+@@ -375,6 +457,7 @@ if(BUILD_SPARSE)
+       CMAKE_COMMAND=${CMAKE_CMD}
+       CUDA_TOOLKIT_ROOT_DIR=${CUDA_TOOLKIT_ROOT_DIR}
+       USE_CUDA=${USE_CUDA}
++      EXTERNAL_DMLC_LIB_PATH=${EXTERNAL_DMLC_LIB_PATH}
+       BINDIR=${CMAKE_CURRENT_BINARY_DIR}
+       INCLUDEDIR="${DGL_INCLUDE_DIRS}"
+       CFLAGS=${CMAKE_C_FLAGS}
+diff --git a/cmake/modules/FindMETIS.cmake b/cmake/modules/FindMETIS.cmake
+new file mode 100644
+index 00000000..67182a58
+--- /dev/null
++++ b/cmake/modules/FindMETIS.cmake
+@@ -0,0 +1,28 @@
++# Find the METIS includes and library
```

Comment:
@bgawrych any ideas how to get this to work in the conda build env? Seems to work fine locally but doesn't detect in the build env.